### ISSUE TITLE
IPolicy break instance creation with disk adoption

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#311]
Originally reported of Google Code with ID 255.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster (ganeti v2.6.0) 2.6.0

<b>What distribution are you using?</b>

Debian (wheezy/sid)

<b>What steps will reproduce the problem?</b>
1.Attempt to create a new instance (gnt-instance add -o image+default -t plain -n vmhost-0  --disk 0:adopt=imagetest,vg=ganeti -B memory=1GB --no-name-check --no-ip-check --no-start --no-install imagetest)

<b>What is the expected output? What do you see instead?</b>

Expected: instance created without issue as the disk being adopted is within the range of accepted disk sizes specified via IPolicy.

Actual: 
Failure: prerequisites not met for this operation:
error type: wrong_input, error details:
Instance allocation to group 6e1b2abd-69c2-4b31-b2df-f96aa5122d76 violates policy: disk-size/0 value 0 is not in range [1024, 1048576]

<b>Please provide any additional information below.</b>

IPolicy:

Instance policy - limits for instances:
  - std
    cpu-count: 1
    disk-count: 1
    disk-size: 1024
    memory-size: 128
    nic-count: 1
    spindle-use: 1
  - max
    cpu-count: 8
    disk-count: 16
    disk-size: 1048576
    memory-size: 32768
    nic-count: 8
    spindle-use: 12
  - min
    cpu-count: 1
    disk-count: 1
    disk-size: 1024
    memory-size: 128
    nic-count: 1
    spindle-use: 1
  - enabled disk templates: sharedfile, diskless, plain, blockdev, drbd, file, rbd
  - vcpu-ratio: 4.0
  - spindle-ratio: 32.0


VG Info:

  --- Logical volume ---
  LV Name                /dev/ganeti/imagetest
  VG Name                ganeti
  LV UUID                Rg2ie1-UgJ1-BK4N-Aae4-0Vci-gIBq-kldkSI
  LV Write Access        read/write
  LV Status              available
  # open                 0
  LV Size                10.00 GiB
  Current LE             2560
  Segments               1
  Allocation             inherit
  Read ahead sectors     auto
  - currently set to     256
  Block device           254:22

```
Originally added on 2012-08-06 18:35:02 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Same here. disk adopting is broken in 2.6. Thank you
```
Originally added on 2012-09-17 16:41:58 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I believe this is because when we adapt disks, we set the size to 0 for the instance creation (because the disk is already there), instead of the actual size of the disk.
We'll need to investigate further.
```
Originally added on 2012-09-18 10:36:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi, is there a work around that we can use?

```
Originally added on 2012-10-19 02:00:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I've committed a fix, ba147ff8 “Fix disk adoption interaction with ipolicy checks”. You can apply the patch locally or wait until it's released as part of 2.6.2.
```
Originally added on 2012-10-22 08:10:39 +0000 UTC.

Added Labels: Type-Defect
Changed State: Fixed
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 09:15:34 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as released the bugs fixed in 2.6.2.
```
Originally added on 2013-01-11 12:13:05 +0000 UTC.

Changed State: Released
